File: src/main/java/toker/warbandscripts/panel/entity/DownloadToken.java
package toker.warbandscripts.panel.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "download_token")
public class DownloadToken {

    private Integer id;
    private String token;
    private String file;
    private PanelUser user;
    private Timestamp time;
    private Boolean isUsed;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @Column(name = "token", nullable = false)
    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    @Column(name = "file", nullable = false)
    @JsonIgnore
    public String getFile() {
        return file;
    }

    public void setFile(String file) {
        this.file = file;
    }

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
    @JsonIgnore
    public PanelUser getUser() {
        return user;
    }

    public void setUser(PanelUser user) {
        this.user = user;
    }

    @Column(name = "time", nullable = false)
    public Timestamp getTime() {
        return time;
    }

    public void setTime(Timestamp time) {
        this.time = time;
    }

    @Column(name = "is_used", nullable = false)
    public Boolean isUsed() {
        return isUsed;
    }

    public void setUsed(Boolean used) {
        isUsed = used;
    }
}

File: src/main/java/toker/panel/entity/DownloadToken.kt
package toker.panel.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import java.sql.Timestamp
import javax.persistence.*

@Entity
@Table(name = "download_token")
class DownloadToken(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id")
        var id: Int? = null,

        @Column(name = "token", nullable = false)
        var token: String? = null,

        @JsonIgnore
        @Column(name = "file", nullable = false)
        var file: String? = null,

        @ManyToOne(fetch = FetchType.EAGER)
        @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
        @JsonIgnore
        var user: PanelUser? = null,

        @Column(name = "time", nullable = false)
        var time: Timestamp? = null,

        @Column(name = "is_used", nullable = false)
        var isUsed: Boolean? = null
) {
        override fun equals(other: Any?): Boolean {
                if (this === other) return true
                if (javaClass != other?.javaClass) return false
                other as DownloadToken
                if (id != other.id) return false
                return true
        }

        override fun hashCode(): Int {
                return id ?: 0
        }
}